Gems/DiffuseProbeGrid/Code/Source/Render/DiffuseProbeGridQueryFullscreenPass.cpp

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Memory/SystemAllocator.h>
+#include <Atom/RHI/CommandList.h>
+#include <Atom/RHI/FrameGraphBuilder.h>
+#include <Atom/RHI/FrameGraphInterface.h>
+#include <Atom/RHI/PipelineState.h>
+#include <Atom/RPI.Public/RenderPipeline.h>
+#include <Atom/RPI.Public/Pass/PassUtils.h>
+#include <Atom/RPI.Public/RPIUtils.h>
+#include <Atom/RPI.Public/Scene.h>
+#include <DiffuseProbeGrid_Traits_Platform.h>
+#include <Render/DiffuseProbeGridQueryFullscreenPass.h>
+#include <Render/DiffuseProbeGridQueryFullscreenPassData.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        RPI::Ptr<DiffuseProbeGridQueryFullscreenPass> DiffuseProbeGridQueryFullscreenPass::Create(const RPI::PassDescriptor& descriptor)
+        {
+            RPI::Ptr<DiffuseProbeGridQueryFullscreenPass> pass = aznew DiffuseProbeGridQueryFullscreenPass(descriptor);
+            return AZStd::move(pass);
+        }
+
+        DiffuseProbeGridQueryFullscreenPass::DiffuseProbeGridQueryFullscreenPass(const RPI::PassDescriptor& descriptor)
+            : RPI::RenderPass(descriptor),
+              m_passDescriptor(descriptor)
+        {
+            if (!AZ_TRAIT_DIFFUSE_GI_PASSES_SUPPORTED)
+            {
+                // GI is not supported on this platform
+                SetEnabled(false);
+            }
+            else
+            {
+                LoadShader();
+            }
+        }
+
+        void DiffuseProbeGridQueryFullscreenPass::LoadShader()
+        {
+            auto passData = RPI::PassUtils::GetPassData<DiffuseProbeGridQueryFullscreenPassData>(m_passDescriptor);
+            bool useAlbedoTexture = (passData && passData->m_useAlbedoTexture);
+
+            // load shader
+            // Note: the shader may not be available on all platforms
+            static const char* QueryFullscreenShaderFilePath = "Shaders/DiffuseGlobalIllumination/DiffuseProbeGridQueryFullscreen.azshader";
```

Comment:
If you're adding hard coded paths, you might also need to update the seedList file for this gem for release builds to work, unless you know that the asset is already referenced by a source asset in such a way that the AssetBundler will find it automatically.